lesson_19/homework19_1.py

```diff
@@ -0,0 +1,114 @@
+"""
+Homework 19.1 Mars Rover “Curiosity” photos.
+
+NASA has an open API that allows retrieving data in JSON format
+about photos taken by the “Curiosity” rover on Mars
+based on specific parameters.
+Among this data, there are links to photos that need to be parsed
+and then downloaded and saved as local files
+(mars_photo1.jpg, mars_photo2.jpg, etc.) using additional requests.
+The task should be implemented using the requests module.
+"""
+
+import logging
+
+import requests
+from requests.exceptions import HTTPError, Timeout
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s',
+)
+_log = logging.getLogger(__name__)
+
+
+class NasaImageParser:
+    """
+    A class discribed a parser for NASA images.
+
+    Attributes:
+        url (str): NASA API link.
+        params (dict): dict of the params.
+    """
+
+    def __init__(self, url: str, params: dict):
+        """Initialize a new NasaImageParser with parameters."""
+        self.base_url = self._clean_url(url)
+        self.params = self._prepare_params(params)
+
+    @staticmethod
+    def _clean_url(url: str) -> str:
+        """Clear the URL if additional chars in the start/end of the string."""
+        if url.startswith('<') and url.endswith('>'):
+            return url[1: -1]
+        return url
+
+    def _prepare_params(self, params: dict) -> dict:
+        """Check if all required params are available."""
+        if not isinstance(params, dict):
+            raise ValueError('Params must be a dictionary.')
+        if 'api_key' not in params:
+            raise ValueError("The 'api_key' parameter is required.")
+        return params
+
+    def get_photos(self):
+        """Get the photos from remote."""
+        try:
+            response = requests.get(
+                self.base_url,
+                params=self.params,
+                timeout=10,
+            )
+            response.raise_for_status()
+            data = response.json()
+            return data.get('photos', [])
+        except (HTTPError, ConnectionError, Timeout) as err:
+            _log.error(f'Error fetching photos: {err}')
+            return []
```

Comment:
we have some logic for get request execution. This logic we use twice - > here and in photo download. So better to create helper function aka get_request and use it in both places